From 2d879ea6e3db220a1273cd5d46ea38610748167e Mon Sep 17 00:00:00 2001
From: "Manjunathappa, Prakash" <prakash.pm@ti.com>
Date: Tue, 10 Jan 2012 16:38:10 +0530
Subject: [PATCH 0675/1092] arm: am33xx: add GPIO backlight control hookup to
 LCD FB driver

Patch adds hookup to toggle GPIO backlight from da8xx-fb LCD frame
buffer driver.

Signed-off-by: Manjunathappa, Prakash <prakash.pm@ti.com>
---
 arch/arm/mach-omap2/board-am335xevm.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/arch/arm/mach-omap2/board-am335xevm.c b/arch/arm/mach-omap2/board-am335xevm.c
index 1b611ff..736b17e 100644
--- a/arch/arm/mach-omap2/board-am335xevm.c
+++ b/arch/arm/mach-omap2/board-am335xevm.c
@@ -140,10 +140,13 @@ static struct lcd_ctrl_config lcd_cfg = {
 	.raster_order		= 0,
 };
 
+static void am335x_gpio_bl_ctrl(int val);
+
 struct da8xx_lcdc_platform_data TFC_S9700RTWV35TR_01B_pdata = {
 	.manu_name		= "ThreeFive",
 	.controller_data	= &lcd_cfg,
 	.type			= "TFC_S9700RTWV35TR_01B",
+	.panel_power_ctrl	= am335x_gpio_bl_ctrl,
 };
 
 #include "common.h"
@@ -845,6 +848,12 @@ static void enable_ecap0(int evm_id, int profile)
 	backlight_enable = true;
 }
 
+static void am335x_gpio_bl_ctrl(int val)
+{
+	/* lcd backlight */
+	gpio_set_value(AM335X_LCD_BL_PIN, val);
+}
+
 static int __init ecap0_init(void)
 {
 	int status = 0;
-- 
1.7.11.2

